### PR TITLE
Reduce usages of Ant

### PR DESCRIPTION
**Pull request by basil**
Replaces usages of Ant with native Java Platform or Jenkins API functionality.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [X] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/jenkins/security/seed/UserSeedChangeListener.java
Comment (basil) [inline core/src/main/java/jenkins/security/seed/UserSeedChangeListener.java:42]: Most likely a mistake from commit 8c490d14c4.

**Review by MarkEWaite — COMMENTED** [review]
Thanks for the improvement.

File: core/src/main/java/hudson/Util.java
Comment (basil) [reply core/src/main/java/hudson/Util.java:1073]: Most of the code base already imports `org.apache.tools.ant.Project` and uses the short form:

```
basil ~/src/jenkinsci/jenkins (master) ❯ rg -li 'import org.apache.tools.ant.Project'
core/src/main/java/hudson/Util.java
core/src/main/java/hudson/FilePath.java
core/src/main/java/hudson/ClassicPluginStrategy.java
core/src/test/java/hudson/FilePathTest.java
core/src/main/java/hudson/util/DirScanner.java
core/src/main/java/jenkins/diagnosis/HsErrPidList.java
core/src/main/java/hudson/lifecycle/WindowsInstallerLink.java
core/src/main/java/jenkins/util/AntClassLoader.java
```

This change is just improving consistency by making the remaining two usages of the long form match the rest of the code base.

File: core/src/main/java/hudson/model/AbstractItem.java
Comment (basil) [reply core/src/main/java/hudson/model/AbstractItem.java:422]: Ditto

**Review by MarkEWaite — APPROVED** [review]
Thanks for the clarification on the import of the Ant `Project` class.  Approved

**Review by MarkEWaite — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
